proposals/dao/tip_110.ts

```diff
@@ -0,0 +1,109 @@
+  const angleEuroRedeemer = await angleEuroRedeemerFactory.deploy();
+  await angleEuroRedeemer.deployed();
+  logging && console.log(`angleEuroRedeemer: ${angleEuroRedeemer.address}`);
+
+  return {
+    angleEuroRedeemer
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // make sure ETH oracle is fresh (for B.AMM not to revert, etc)
+  // Read Chainlink ETHUSD price & override chainlink storage to make it a fresh value
+  const ethPrice = (await contracts.chainlinkEthUsdOracleWrapper.read())[0].toString() / 1e10;
+  await overwriteChainlinkAggregator(addresses.chainlinkEthUsdOracle, Math.round(ethPrice).toString(), '8');
+
+  // read pcvStats before proposal execution
+  pcvStatsBefore = await contracts.collateralizationOracle.pcvStats();
+
+  // angle multisig action : make enough USDC collateral available for redemptions
+  const angleMultisigSigner = await getImpersonatedSigner('0x0C2553e4B9dFA9f83b1A6D3EAB96c4bAaB42d430');
+  await forceEth(angleMultisigSigner.address);
+  await contracts.anglePoolManagerUsdc.connect(angleMultisigSigner).updateStrategyDebtRatio(
+    addresses.angleStrategyUsdc1, // USDC strategy has 57M deployed
+    '0'
+  );
+  await contracts.angleStrategyUsdc1.harvest();
+
+  // read DAI PSM balance before proposal execution
+  daiBalanceBefore = await contracts.dai.balanceOf(addresses.daiFixedPricePSM);
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // check ANGLE token movement
+  expect(await contracts.angle.balanceOf(addresses.angleDelegatorPCVDeposit)).to.be.equal('0');
+  expect(await contracts.angle.balanceOf(addresses.tribalCouncilSafe)).to.be.at.least(
+    ethers.utils.parseEther('200000')
+  );
+  // check deposit is empty
+  expect(await contracts.agEurUniswapPCVDeposit.balance()).to.be.equal('0');
+  expect(await contracts.fei.balanceOf(addresses.agEurUniswapPCVDeposit)).to.be.equal('0');
+  expect(await contracts.agEUR.balanceOf(addresses.agEurUniswapPCVDeposit)).to.be.equal('0');
+
+  // check redemptions of agEUR > DAI
+  const daiRedeemed = (await contracts.dai.balanceOf(addresses.daiFixedPricePSM)).sub(daiBalanceBefore);
+  console.log('daiRedeemed', daiRedeemed.toString() / 1e18);
+  expect(daiRedeemed).to.be.at.least(ethers.utils.parseEther('9500000')); // >9.5M DAI
```

Comment:
the expected behavior, if enough USDC is available for redemptions, is to convert all agEUR to DAI with a fixed fee of 0.45% (Angle Protocol redemption) + 0% (Maker USDC->DAI PSM)